xdsl/dialects/arith.py

```diff
@@ -73,14 +74,21 @@ def from_attr(attr: Attribute, typ: Attribute) -> Constant:
         return Constant.create(result_types=[typ], attributes={"value": attr})
 
     @staticmethod
-    def from_int_constant(val: Union[int, Attribute],
-                          typ: Union[int, Attribute]) -> Constant:
+    def from_int(val: Union[int, Attribute],
+                 typ: Union[int, Attribute]) -> Constant:
         if isinstance(typ, int):
             typ = IntegerType.from_width(typ)
         return Constant.create(
             result_types=[typ],
             attributes={"value": IntegerAttr.from_params(val, typ)})
 
+    @staticmethod
+    def from_float(val: Union[float, Attribute],
```

Comment:
Dito for this. The type really only describes the bitwidth of the float, right? Therefore, this could also be `from_float_and_width`.